packages/grafana-ui/src/components/Select/SelectBase.tsx

```diff
@@ -285,23 +285,26 @@ export function SelectBase<T, Rest = {}>({
 
     // code needed to allow editing a custom value once entered
     // we only want to do this for single selects, not multi
-    if (!isMulti) {
+    // and only if the inputValue is uncontrolled
+    if (!isMulti && inputValue === undefined) {
       creatableProps.inputValue = internalInputValue;
       creatableProps.onMenuOpen = () => {
         // make sure we call the base onMenuOpen if it exists
         commonSelectProps.onMenuOpen?.();
 
-        // restore the input state to the selected value
-        setInternalInputValue(selectedValue?.[0]?.label ?? '');
+        // restore the input state to the selected value if it's custom
+        if (!options.find((o) => o.value === selectedValue?.[0]?.value)) {
+          setInternalInputValue(selectedValue?.[0]?.label ?? '');
+        }
```

Comment:
not sure this method is going to work, with https://github.com/grafana/scenes/pull/763 the options I pass to the select component is not the full list of options but can be a subset (if above 10K).

Reply:
@torkelo hmmmm, not sure how to differentiate between a created custom option and one that just isn't part of the subset then 🤔 

since you're looping through all the values in `getOptionSearcher` anyway, can we change the [implementation when the search string is empty](https://github.com/grafana/scenes/pull/763/files#diff-1ea2ba24614b5ca50be489c000e394f1e0199cdfb0062919865fefca84186f4aR28-R34) to guarantee that we pass in the selected option in the subset if it exists in the options? maybe something like:
```tsx
  let currentIndex = -1;

  if (!Array.isArray(value)) {
   currentIndex = options.findIndex((x) => x.value === value);
    if (currentIndex === -1) {
      allOptions = [{ value: value, label: String(text) }, ...allOptions];
    }
  }
  
  return (search: string) => {
    if (search === '') {
      if (allOptions.length > limit) {
        return currentIndex < limit ? allOptions.slice(0, limit) : [...allOptions.slice(0, limit), allOptions[currentIndex]];
      } else {
        return allOptions;
      }
    }
  }
```

that way the selected option is guaranteed to be part of the initial subset, and this logic will still work?

or we can use `onCreateOption` with some state to track the newly created options internally, but that will only work if the option has just been created now. it won't work if you load the page with a custom value preselected then open the select. 

or open to other suggestions 🤔

Reply:
@ashharrison90 I think if we revert https://github.com/grafana/scenes/pull/764 and just add that behavior to scenes VariableSelect the needed code change is very small, https://github.com/grafana/scenes/pull/764

Reply:
sorry linked to wrong PR (mean we should revert the PR that added preserve custom value to core Select)

Reply:
yep got it 👍 

so i think the plan would end up being:
- close this
- merge https://github.com/grafana/scenes/pull/764 into https://github.com/grafana/scenes/pull/763
- merge https://github.com/grafana/scenes/pull/763 into scenes main
- pull in latest scenes in `grafana/grafana`
- revert https://github.com/grafana/grafana/pull/87843

that way we don't have any weird inbetween versions where the behaviour doesn't work properly.